ui/blocks/AffiliateCard.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react'
+import { Affiliate, findAffiliateByCode } from '~lib/api/affiliates'
+import { Card } from '~ui/core'
+
+interface Props {
+  code: string
+}
+
+export const AffiliateCard: React.FC<Props> = ({ code }) => {
+  const [affiliate, setAffiliate] = React.useState<Affiliate>(null)
+
+  React.useEffect(() => {
+    if (code === undefined) return
+    findAffiliateByCode(code)
+      .then((affiliate) => setAffiliate(affiliate))
+      .catch(() => {
+        setAffiliate(null)
+      })
+  }, [code])
+
+  return affiliate ? (
+    <Card>
+      <a href={affiliate.logoLink} target="_blank" rel="noreferrer noopener">
```

Comment:
As I wrote in https://github.com/railslove/rcvr-api/pull/66#discussion_r589311675
We either should require logo_link to always be set or check for it here and not create a link if it's empty.

Reply:
Thing is, if the link is null, the link is just not clickable which is the behaviour we want. What you think?

Reply:
I wouldn't rely on consistent behavior on all browsers. Some browsers might think that it's a relative path and practically reload the page. It's semantically incorrect to disable the link in this way (think of WCAG).

Reply:
ok got you.

Reply:
@salimhb see my latest commit.